osu.Game.Rulesets.Catch/Objects/Drawables/DropletPiece.cs

```diff
@@ -23,7 +23,7 @@ public DropletPiece()
         private void load(DrawableHitObject drawableObject)
         {
             DrawableCatchHitObject drawableCatchObject = (DrawableCatchHitObject)drawableObject;
-            var hitObject = drawableCatchObject.HitObject;
+            var hitObject = (PalpableCatchHitObject)drawableCatchObject.HitObject;
```

Comment:
I'd do this slightly differently, namely re-expose the HO as the more-specific type in the DHO:

```csharp
public abstract class DrawablePalpableCatchHitObject : DrawableCatchHitObject
{
    public new PalpableCatchHitObject HitObject => (PalpableCatchHitObject)base.HitObject;
    // ...
}
```

which then can be used more directly here to cast once instead of twice:

```csharp
var drawablePalpable = (DrawablePalpableHitObject)drawableObject;
var hitObject = drawablePalpable.HitObject;
```

Reply:
(To all comments about `DrawablePalpableHitObject` casts in `load`)
I agree with suggestions, but why I left ugly casts there is because those codes will be rewritten when the hit objects are pooled.